pandas/core/arrays/period.py

```diff
@@ -141,21 +142,27 @@ def _simple_new(cls, values, freq=None, **kwargs):
         Ordinals in an ndarray are fastpath-ed to `_from_ordinals`
         """
 
+        if is_period_dtype(values):
+            freq = dtl.validate_dtype_freq(values.dtype, freq)
+            values = values.asi8
+
         if not is_integer_dtype(values):
             values = np.array(values, copy=False)
             if len(values) > 0 and is_float_dtype(values):
                 raise TypeError("{cls} can't take floats"
                                 .format(cls=cls.__name__))
-            return cls(values, freq=freq)
+            return cls(values, freq=freq, **kwargs)
 
-        return cls._from_ordinals(values, freq)
+        return cls._from_ordinals(values, freq=freq, **kwargs)
```

Comment:
was this causing bugs that kwargs was not passed?

Reply:
This is necessary in order to allow `PeriodIndex` to inherit `_simple_new` from `PeriodArrayMixin`.

The PeriodIndex constructors in particular (but also DatetimeIndex and TimedeltaIndex) are really messy and complicated.  Inheriting methods and deleting as much as possible makes it easier for me to simplify them, even if some of that will end up getting copy/pasted down the road if/when inheritance is replaced.

Reply:
Can you then move the implementation and share it by calling the other class instead of inheriting?

Reply:
Do you mean moving the `_from_ordinals` implementation?  We need both because the Index version sets self.name and calls _reset_identity

Reply:
I meant in PeriodIndex._simple_new calling PeriodArrayMixin._simple_new, similar as you did in one of the previous PRs (didn't look at the code, so don't know if that is possible in this case)

Reply:
I think I get what you're saying, will give it a shot

Reply:
Did anything come of this, one way or the other?

Reply:
Yes.  In an earlier commit I deleted PeriodIndex._simple_new and retained PeriodIndex._from_ordinals.  Now that has been reversed, and its non-trivially nicer.